src/wp-includes/admin-bar.php

```diff
@@ -287,9 +287,8 @@ function wp_admin_bar_my_account_item( $wp_admin_bar ) {
 			'title'  => $howdy . $avatar,
 			'href'   => $profile_url,
 			'meta'   => array(
-				'class'      => $class,
-				/* translators: %s: Current user's display name. */
-				'menu_title' => sprintf( __( 'Howdy, %s' ), $current_user->display_name ),
+				'class'      => ( false === $avatar ) ? '' : 'with-avatar',
```

Comment:
I do not recommend changing `empty()`.

The default `get_avatar()` function returns a string or `false`, but
1. The `get_avatar` filter technically can return an empty string.
2. The function is pluggable, and a custom replacement could return anything (though if it is not a string or `false`, that would cause a problem with the node `title`).

Reply:
Reverted in https://github.com/WordPress/wordpress-develop/pull/10109/commits/100f2ab2ccfccaf28759b172720cab1916431213